infra/examples-dev/aws-all/main.tf

```diff
@@ -0,0 +1,161 @@
+terraform {
+  required_providers {
+    # for the infra that will host Psoxy instances
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.12"
+    }
+  }
+
+  # we recommend you use a secure location for your Terraform state (such as S3 bucket), as it
+  # may contain sensitive values (such as API keys) depending on which data sources you configure.
+  #
+  # local may be safe for production-use IFF you are executing Terraform from a secure location
+  #
+  # Please review and seek guidance from your Security team if in doubt.
+  backend "local" {
+  }
+
+  # example remove backend (this S3 bucket must already be provisioned, and AWS role executing
+  # terraform must be able to read/write to it - and use encryption key, if any)
+  #  backend "s3" {
+  #    bucket = "terraform_state_bucket" # fill with S3 bucket where you want the statefile to be
+  #    key    = "prod_state" # fill with path where you want state file to be stored
+  #    region = "us-east-1" # cannot be a variable
+  #  }
+}
+
+
+## Data Sources configuration
+## (eg, sources you want to connect to Worklytics)
+
+# general cases
+module "worklytics_connectors" {
+  source = "../../modules/worklytics-connectors"
+
+  enabled_connectors    = var.enabled_connectors
+  example_jira_issue_id = var.example_jira_issue_id
+  jira_cloud_id         = var.jira_cloud_id
+  jira_server_url       = var.jira_server_url
+  salesforce_domain     = var.salesforce_domain
+}
+
+# sources which require additional dependencies are split into distinct Terraform files, following
+# the naming convention of `{source-identifier}.tf`, eg `msft-365.tf`
+# lines below merge results of those files back into single maps of sources
+
+locals {
+  api_connectors = merge(
+    module.worklytics_connectors.enabled_api_connectors,
+    module.worklytics_connectors_google_workspace.enabled_api_connectors,
+    local.msft_api_connectors_with_auth
+  )
+}
+
+
+locals {
+  bulk_connectors = merge(
+    module.worklytics_connectors.enabled_bulk_connectors,
+    var.custom_bulk_connectors,
+  )
+}
+
+
+## Host platform (AWS) configuration
+
+# NOTE: you need to provide credentials. usual way to do this is to set env vars:
+#        AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY
+# see https://registry.terraform.io/providers/hashicorp/aws/latest/docs#authentication for more
+# information as well as alternative auth approaches
+provider "aws" {
```

Comment:
If following suggestion move to  `providers.tf`

Reply:
imho, I think better to organize it as an `aws.tf` right? as the `aws` provider is coupled to the "feature" of hosting in AWS. 

But yeah, doesn't exactly extend well to GCP case, as we'd end up with `google` provider defined in both the `gcp.tf` for the host as well as potentially a `google-workspace.tf` for the data source.